src/Tribe/Views/V2/View.php

```diff
@@ -1349,4 +1369,32 @@ protected function setup_messages( array $events ) {
 			}
 		}
 	}
+
+	/**
+	 * Returns whether the View page should be reset or not.
+	 *
+	 * The View page should be reset when the View or filtering parameters that are not the page change.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool Whether the View page should be reset or not.
+	 */
+	protected function should_reset_page() {
+		if ( null === $this->should_reset_page ) {
+			$prev_url    = $this->context->get( 'view_prev_url', '' );
+			$current_url = $this->context->get( 'view_url', '' );
+
+			$view_data = $this->context->get( 'view_data', [] );
+			$bar_data = array_filter( $view_data, static function ( $value, $key ) {
+				return 0 === strpos( $key, 'tribe-bar-' ) && ! empty( $value );
+			}, ARRAY_FILTER_USE_BOTH );
```

Comment:
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: e52ba4746711e3447b1468b7c491ad5d</sub>